Opencascade "Cannot open include file: 'Standard.hxx'"

**Describe the bug**
Opencascade 7.5.0 builds (as windows-x64-static) without any errors. 
However Visual Studio 2019 can not find the header files for Opencascade itself whenever a header from said library is included.
For example when I want to `#include <opencascade/gp.hxx>`, Visual Studio 2019 says:
`1>C:\Sys\vcpkg\installed\x64-windows-static\include\opencascade\gp.hxx(20,10): fatal error C1083: Cannot open include file: 'Standard.hxx': No such file or directory`

**Environment**
- OS: Windows 10
- Compiler: Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29337 for x86

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install opencascade.
2. Create empty project (C++) in Visual Studio 2019.
3. Create a .cpp file and `#include <opencascade/gp.hxx>`.
4. See error: `1>C:\Sys\vcpkg\installed\x64-windows-static\include\opencascade\gp.hxx(20,10): fatal error C1083: Cannot open include file: 'Standard.hxx': No such file or directory`.

--- a/ports/opencascade/CONTROL
+++ b/ports/opencascade/CONTROL
@@ -1,5 +1,6 @@
 Source: opencascade
 Version: 7.5.0
+Port-Version: 1
 Build-Depends: freetype
 Description: Open CASCADE Technology (OCCT) is an open-source software development platform for 3D CAD, CAM, CAE.
 Supports: !(uwp|osx|linux|arm)

--- a/ports/opencascade/fix-install-prefix-path.patch
+++ b/ports/opencascade/fix-install-prefix-path.patch
@@ -1,12 +1,15 @@
 diff --git a/adm/templates/OpenCASCADEConfig.cmake.in b/adm/templates/OpenCASCADEConfig.cmake.in
-index 4937103b..cd35e07d 100644
+index e391ddf..c926c49 100644
 --- a/adm/templates/OpenCASCADEConfig.cmake.in
 +++ b/adm/templates/OpenCASCADEConfig.cmake.in
-@@ -23,6 +23,7 @@ set (OpenCASCADE_DEVELOPMENT_VERSION "@OCC_VERSION_DEVELOPMENT@")
- # - in Windows style: $INSTALL_DIR/cmake
+@@ -24,6 +24,7 @@ set (OpenCASCADE_DEVELOPMENT_VERSION "@OCC_VERSION_DEVELOPMENT@")
+ # - in Android style: $INSTALL_DIR/libs/$CMAKE_ANDROID_ARCH_ABI/cmake/opencascade-<version>
  get_filename_component (OpenCASCADE_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
  get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 +get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
  if (OpenCASCADE_INSTALL_PREFIX MATCHES "/cmake$")
    get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
  endif()
+-- 
+2.14.3.windows.1
+

--- a/ports/opencascade/fix-install-prefix-path.patch
+++ b/ports/opencascade/fix-install-prefix-path.patch
@@ -1,0 +1,12 @@
+diff --git a/adm/templates/OpenCASCADEConfig.cmake.in b/adm/templates/OpenCASCADEConfig.cmake.in
+index 4937103b..cd35e07d 100644
+--- a/adm/templates/OpenCASCADEConfig.cmake.in
++++ b/adm/templates/OpenCASCADEConfig.cmake.in
+@@ -23,6 +23,7 @@ set (OpenCASCADE_DEVELOPMENT_VERSION "@OCC_VERSION_DEVELOPMENT@")
+ # - in Windows style: $INSTALL_DIR/cmake
+ get_filename_component (OpenCASCADE_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+ get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
++get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
+ if (OpenCASCADE_INSTALL_PREFIX MATCHES "/cmake$")
+   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
+ endif()

--- a/ports/opencascade/install-include-dir.patch
+++ b/ports/opencascade/install-include-dir.patch
@@ -1,0 +1,32 @@
+From 32c4bdd88555309752215c53842d438cb51bcb62 Mon Sep 17 00:00:00 2001
+From: bloess <josua.bloess@uni-bayreuth.de>
+Date: Mon, 15 Feb 2021 16:26:36 +0100
+Subject: [PATCH] install include-dir
+
+---
+ CMakeLists.txt | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 100d613..464f95a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1171,6 +1171,15 @@ foreach (OCCT_MODULE ${OCCT_MODULES})
+ endforeach()
+ set (OCCT_MODULES_ENABLED ${OCCT_MODULES_ENABLED_SORTED})
+ 
++foreach (OCCT_LIBRARY ${OCCT_LIBRARIES})
++	target_include_directories(
++		${OCCT_LIBRARY} 
++		INTERFACE
++		#$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source>
++		$<INSTALL_INTERFACE:include/opencascade>)
++
++endforeach()
++
+ # Add all targets to the build-tree export set
+ export (TARGETS ${OCCT_LIBRARIES} FILE "${CMAKE_BINARY_DIR}/OpenCASCADETargets.cmake")
+ 
+-- 
+2.14.3.windows.1
+

--- a/ports/opencascade/install-include-dir.patch
+++ b/ports/opencascade/install-include-dir.patch
@@ -11,7 +11,7 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
 index 100d613..464f95a 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -1171,6 +1171,15 @@ foreach (OCCT_MODULE ${OCCT_MODULES})
+@@ -1171,6 +1171,14 @@ foreach (OCCT_MODULE ${OCCT_MODULES})
  endforeach()
  set (OCCT_MODULES_ENABLED ${OCCT_MODULES_ENABLED_SORTED})
  
@@ -19,7 +19,6 @@ index 100d613..464f95a 100644
 +	target_include_directories(
 +		${OCCT_LIBRARY} 
 +		INTERFACE
-+		#$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source>
 +		$<INSTALL_INTERFACE:include/opencascade>)
 +
 +endforeach()

--- a/ports/opencascade/install-include-dir.patch
+++ b/ports/opencascade/install-include-dir.patch
@@ -19,7 +19,7 @@ index 100d613..464f95a 100644
 +	target_include_directories(
 +		${OCCT_LIBRARY} 
 +		INTERFACE
-+		$<INSTALL_INTERFACE:include/opencascade>)
++		$<INSTALL_INTERFACE:include>)
 +
 +endforeach()
 +

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -47,6 +47,27 @@ vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/opencascade)
 
+#make occt includes relative to source_file
+list(APPEND ADDITIONAL_HEADERS 
+      "ExprIntrp.tab.h"
+	  "FlexLexer.h"
+	  "glext.h"
+	  "igesread.h"
+	  "NCollection_Haft.h"
+	  "OSD_PerfMeter.h"
+	  "Standard_values.h"
+    )
+
+file(GLOB files "${CURRENT_PACKAGES_DIR}/include/opencascade/[a-zA-Z0-9_]*\.[hgl]xx")
+foreach(file_name ${files})
+	file(READ ${file_name} filedata)
+	string(REGEX REPLACE "# *include \<([a-zA-Z0-9_]*\.[hgl]xx)\>" "#include \"\\1\"" filedata "${filedata}")
+	foreach(extra_header ${ADDITIONAL_HEADERS})
+		string(REGEX REPLACE "# *include \<${extra_header}\>" "#include \"${extra_header}\"" filedata "${filedata}")
+	endforeach()
+	file(WRITE ${file_name} "${filedata}")
+endforeach()
+
 # Remove libd to lib, libd just has cmake files we dont want too
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib)
 file(RENAME ${CURRENT_PACKAGES_DIR}/debug/libd ${CURRENT_PACKAGES_DIR}/debug/lib)

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -8,7 +8,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-pdb-find.patch
-		install-include-dir.patch
+        install-include-dir.patch
 )
 
 if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
@@ -110,5 +110,4 @@ else()
 endif()
 
 file(INSTALL ${SOURCE_PATH}/OCCT_LGPL_EXCEPTION.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-
 

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -67,7 +67,6 @@ foreach(file_name ${files})
 	file(WRITE ${file_name} "${filedata}")
 endforeach()
 
-
 # Remove libd to lib, libd just has cmake files we dont want too
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib)
 file(RENAME ${CURRENT_PACKAGES_DIR}/debug/libd ${CURRENT_PACKAGES_DIR}/debug/lib)

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -46,27 +46,6 @@ vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/opencascade)
 
-#make occt includes relative to source_file
-list(APPEND ADDITIONAL_HEADERS 
-      "ExprIntrp.tab.h"
-	  "FlexLexer.h"
-	  "glext.h"
-	  "igesread.h"
-	  "NCollection_Haft.h"
-	  "OSD_PerfMeter.h"
-	  "Standard_values.h"
-    )
-
-file(GLOB files "${CURRENT_PACKAGES_DIR}/include/opencascade/[a-zA-Z0-9_]*\.[hgl]xx")
-foreach(file_name ${files})
-	file(READ ${file_name} filedata)
-	string(REGEX REPLACE "#include \<([a-zA-Z0-9_]*\.[hgl]xx)\>" "#include \"\\1\"" filedata "${filedata}")
-	foreach(extra_header ${ADDITIONAL_HEADERS})
-		string(REGEX REPLACE "#include \<${extra_header}\>" "#include \"${extra_header}\"" filedata "${filedata}")
-	endforeach()
-	file(WRITE ${file_name} "${filedata}")
-endforeach()
-
 # Remove libd to lib, libd just has cmake files we dont want too
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib)
 file(RENAME ${CURRENT_PACKAGES_DIR}/debug/libd ${CURRENT_PACKAGES_DIR}/debug/lib)

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -8,7 +8,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-pdb-find.patch
-		fix-install-prefix-path.patch
+        fix-install-prefix-path.patch
         install-include-dir.patch
 )
 

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -59,7 +59,6 @@ list(APPEND ADDITIONAL_HEADERS
 	  "Standard_values.h"
     )
 
-
 file(GLOB files "${CURRENT_PACKAGES_DIR}/include/opencascade/[a-zA-Z0-9_]*\.[hgl]xx")
 foreach(file_name ${files})
 	file(READ ${file_name} filedata)
@@ -111,5 +110,5 @@ else()
 endif()
 
 file(INSTALL ${SOURCE_PATH}/OCCT_LGPL_EXCEPTION.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-#file(INSTALL ${CURRENT_PACKAGES_DIR}/include/opencascade/ DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+
 

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -44,9 +44,7 @@ vcpkg_configure_cmake(
 
 vcpkg_install_cmake()
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/opencascade TARGET_PATH share/opencascade)
-
-
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/opencascade)
 
 #make occt includes relative to source_file
 list(APPEND ADDITIONAL_HEADERS 
@@ -110,4 +108,3 @@ else()
 endif()
 
 file(INSTALL ${SOURCE_PATH}/OCCT_LGPL_EXCEPTION.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-pdb-find.patch
+		install-include-dir.patch
 )
 
 if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
@@ -44,6 +45,31 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/opencascade TARGET_PATH share/opencascade)
+
+
+
+#make occt includes relative to source_file
+list(APPEND ADDITIONAL_HEADERS 
+      "ExprIntrp.tab.h"
+	  "FlexLexer.h"
+	  "glext.h"
+	  "igesread.h"
+	  "NCollection_Haft.h"
+	  "OSD_PerfMeter.h"
+	  "Standard_values.h"
+    )
+
+
+file(GLOB files "${CURRENT_PACKAGES_DIR}/include/opencascade/[a-zA-Z0-9_]*\.[hgl]xx")
+foreach(file_name ${files})
+	file(READ ${file_name} filedata)
+	string(REGEX REPLACE "#include \<([a-zA-Z0-9_]*\.[hgl]xx)\>" "#include \"\\1\"" filedata "${filedata}")
+	foreach(extra_header ${ADDITIONAL_HEADERS})
+		string(REGEX REPLACE "#include \<${extra_header}\>" "#include \"${extra_header}\"" filedata "${filedata}")
+	endforeach()
+	file(WRITE ${file_name} "${filedata}")
+endforeach()
+
 
 # Remove libd to lib, libd just has cmake files we dont want too
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib)
@@ -85,3 +111,5 @@ else()
 endif()
 
 file(INSTALL ${SOURCE_PATH}/OCCT_LGPL_EXCEPTION.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+#file(INSTALL ${CURRENT_PACKAGES_DIR}/include/opencascade/ DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+

--- a/ports/opencascade/portfile.cmake
+++ b/ports/opencascade/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-pdb-find.patch
+		fix-install-prefix-path.patch
         install-include-dir.patch
 )
 

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4282,7 +4282,7 @@
     },
     "opencascade": {
       "baseline": "7.5.0",
-      "port-version": 0
+      "port-version": 1
     },
     "opencc": {
       "baseline": "2020-04-26",

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "751d9ef5966ac3f08574de620b1544b5131a4bc6",
+      "git-tree": "1afbbcf0d6d85963a334b8ae8e95b4fe16a9ff5b",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "5d83e195432acad4c28a14aa033e7d97be8ac071",
+      "git-tree": "8405418fb419e26995ba8a59ed7df88240d4cdfd",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "5d83e195432acad4c28a14aa033e7d97be8ac071",
+      "version-string": "7.5.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "0d003408ab69f77a4ca2c02de115087c87a46b27",
       "version-string": "7.5.0",
       "port-version": 0

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "8405418fb419e26995ba8a59ed7df88240d4cdfd",
+      "git-tree": "c02c43ea384c06aea712e10d5f32bb7b386944da",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "4b4e05154cb8135c68c5da9caf23417a5d45d123",
+      "git-tree": "ca701bcf8acc1652cfa2d2b384eb128da1a9944f",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "6c021c104a573841d1d49bb9358453c6b28c2aff",
+      "git-tree": "751d9ef5966ac3f08574de620b1544b5131a4bc6",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ca701bcf8acc1652cfa2d2b384eb128da1a9944f",
+      "git-tree": "6c021c104a573841d1d49bb9358453c6b28c2aff",
       "version-string": "7.5.0",
       "port-version": 1
     },

--- a/versions/o-/opencascade.json
+++ b/versions/o-/opencascade.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c02c43ea384c06aea712e10d5f32bb7b386944da",
+      "git-tree": "4b4e05154cb8135c68c5da9caf23417a5d45d123",
       "version-string": "7.5.0",
       "port-version": 1
     },

